yatta/src/main.rs

```diff
@@ -321,10 +314,12 @@ fn handle_windows_event_message(mut ev: WindowsEvent, desktop: Arc<Mutex<Desktop
                         }
                     }
                 }
+            } else if !display.get_workspace().windows.contains(&ev.window) {
```

Comment:
I have been playing around with this part trying to get the following situation working:

Workspace 1 has an application open with a clickable link
Workspace 2 has Firefox
When I click the link in the application running on Workspace 1
Then I want to be directed to the newly opened tab in Firefox in Workspace 2

This code gets me to the right workspace, but it won't yet reliably ensure that the browser is the focused window:

```rust
} else if !display.get_workspace().windows.contains(&ev.window) {
    let mut workspace_idx = None;

    for (i, w) in display.workspaces.iter().enumerate() {
        if w.windows.contains(&ev.window) {
            workspace_idx = Option::from(i);
        }
    }

    if let Some(w_idx) = workspace_idx {
        display.set_workspace(w_idx);
    }
}
```

Reply:
Unfortunately, this solution reintruduces a couple of problems that I fixed by adding a window.hide in there: Some programs (like Firefox) when hidden sometimes just don't hide or reshow immediately, and this code makes these programs just sometimes immediately switch back to their workspace when switching away from them.

Reply:
I mean, given that problem, there might be no way to implement this at all.

Reply:
I have modified that snippet a little more to invoke the hiding when a workspace index can't be determined:

```rust
match workspace_idx {
    None => ev.window.hide(),
    Some(idx) => display.set_workspace(idx),
}
```

As far as I can tell this hasn't introduced any issues with hiding and reshowing with Firefox across workspaces and allows for following the focus of links opened in a browser that is in a different workspace.

When you get a chance can you try running a build with this for a little while and reporting how it goes for you?

I'm also going to keep this build running for the rest of the day and I'll report if any related issues arise.

Reply:
Hmm... While this doesn't seem like it would fix the issue, the fact that you haven't experienced this issue makes me think that it might have been caused by something else. I will try this version and tell if I have experienced any issues with it.

Reply:
> 
> 
> I have been playing around with this part trying to get the following situation working:
> 
> Workspace 1 has an application open with a clickable link
> Workspace 2 has Firefox
> When I click the link in the application running on Workspace 1
> Then I want to be directed to the newly opened tab in Firefox in Workspace 2
> 
> This code gets me to the right workspace, but it won't yet reliably ensure that the browser is the focused window:
> 
> ```rust
> } else if !display.get_workspace().windows.contains(&ev.window) {
>     let mut workspace_idx = None;
> 
>     for (i, w) in display.workspaces.iter().enumerate() {
>         if w.windows.contains(&ev.window) {
>             workspace_idx = Option::from(i);
>         }
>     }
> 
>     if let Some(w_idx) = workspace_idx {
>         display.set_workspace(w_idx);
>     }
> }
> ```

Just FYI there is an `iter.position` command, which can be used to write things like these in one line.

Reply:
I tried implementing your code, but as I sometimes instead of switching to another workspace other windows would now sometimes switch to the current workspace when switching from a workspace with firefox in it.

Reply:
Do you think you can come up with a minimal reproducible example of a start state of workspaces and a sequence of events that you can use to reliably produce the behaviour you're talking about? If you can do a little screen recording of it as well that would be amazing. 

I haven't seen any funky today, but it might be that I am just not registering the same thing that you are seeing as weird behaviour. 😅

Reply:
That seems like a good idea and I will try to do so tomorrow, but it might not be easy as it doesn't happen often and seems to just be random.

Reply:
Also, it only happens with Firefox.

Reply:
It happened to me this morning! Though I could not tell you for the life of me the exact sequence of events that produced the behaviour. I'm happy to leave this as-is for now and open a separate issue for trying to get focus changes to switch to other workspaces when applicable in the future.